src/components/resource/edit-custom-title/resource-edit-custom-title.js

```diff
@@ -41,15 +41,24 @@ class ResourceEditCustomTitle extends Component {
 
   state = {
     resourceSelected: this.props.initialValues.isSelected,
-    resourceHidden: this.props.initialValues.isHidden,
+    resourceVisible: this.props.initialValues.isVisible,
     showSelectionModal: false,
     allowFormToSubmit: false,
     formValues: {}
   }
 
   componentWillReceiveProps(nextProps) {
     let wasPending = this.props.model.update.isPending && !nextProps.model.update.isPending;
-    let needsUpdate = !isEqual(this.props.initialValues, nextProps.initialValues);
+    let needsUpdate = !isEqual(this.props.model, nextProps.model);
+
+    if ((nextProps.initialValues.isSelected !== this.props.initialValues.isSelected) ||
+    (nextProps.initialValues.isVisible !== this.props.initialValues.isVisible)) {
+      this.setState({
+        ...this.state,
+        resourceSelected: nextProps.initialValues.isSelected,
+        resourceVisible: this.props.initialValues.isVisible
```

Comment:
Should this be nextProps.initialValues.isVisible? I am missing why these both can be different where resourceSelected: nextProps.... where as resourceVisible: this.props....

Reply:
absolutely right -- good eye!